lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -418,19 +418,28 @@ class CourseEnrollmentDetailsView(APIView):
 
         This api works with all versions {api_version}, you can use: v0.5, v1, v2 or v3
 
-        GET /api/mobile/{api_version}/course_info/{course_id}}/enrollment_details
+        GET /api/mobile/{api_version}/course_info/{course_id}/enrollment_details
 
     """
-    @mobile_course_access()
-    def get(self, request, course, *args, **kwargs):
+    def get(self, request, *args, **kwargs):
         """
         Handle the GET request
 
         Returns user enrollment and course details.
         """
+        course_key_string = kwargs.get('course_id')
+        if not course_key_string:
+            return Response(data={'error': 'course_id is required.'}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
Nit: Would this make more sense to be a 400 (Bad Request) rather than Not Found?

The problem with the request is that required data is missing from the request, not so much that the resource on the server side is missing?